apps/services/user-profile/src/app/v2/user-profile.service.ts

```diff
@@ -16,10 +16,15 @@ import { UserProfileDto } from './dto/user-profile.dto'
 import { IslykillService } from './islykill.service'
 import { DataStatus } from '../user-profile/types/dataStatusTypes'
 import { NudgeType } from '../types/nudge-type'
+import { ClientType } from '../types/ClientType'
 
 export const NUDGE_INTERVAL = 6
 export const SKIP_INTERVAL = 1
 
+export const MIGRATION_DATE = new Date(
+  process.env.USER_PROFILE_MIGRATION_DATE ?? '2022-01-01',
+)
```

Comment:
To be consistent we should add `migrationDate` to `environment.ts` where we initialise it from the env variable and have dev fallback value but make the app startup fail in production if it is missing. So better yet would be to refactor the environment.ts setup to a Config and use `env.required` with dev fallback value.